### PR TITLE
GI-84 & GI-90: Implementation Quiz Level 2

### PR DESCRIPTION
**Pull request by noadjamila**
## Description

Implementation of the Quiz Level 2. Includes frontend and backend implementation.
Frontend fetches the question of level 2 from the backend and displays them with three answer options (yes, no, skip). A Progressbar and a small animation of the radio buttons are also implemented. A Riasec-score is calculated based on the users answers and the top three of them is send to the backend. The backend then goes through the database to return the ids of studys fitting to the three top riasec types. 

## Jira Ticket

Link to the Jira ticket: 
[GI-84](https://techstudyfinder.atlassian.net/browse/GI-84)
[GI-90](https://techstudyfinder.atlassian.net/browse/GI-90)

## Author Checklist

- [ ] Code follows project coding standards and conventions
- [ ] Tests have been written or updated
- [ ] All tests pass locally
- [ ] No sensitive data or secrets are included
- [ ] Documentation (README, comments) has been updated if necessary

## Reviewer Notes

Please verify the following during review:

- The implementation matches the described requirements and Jira ticket
- Code changes are logically correct and maintainable
- No unnecessary dependencies or side effects introduced
- Tests cover critical logic and edge cases
- Naming, structure, and readability meet team standards

---

_Please ensure all criteria are met before merging._

### CODE REVIEW

**Comment by DamarisLi** [comment]
Richtig gut bisher! Wir sollten aber wahrscheinlich schon jetzt die backend layer trennen (separation of concerns) mit controller, service und repository. Dann ist es auch nachher einfacher, andere PRs zu mergen :)

File: client/src/pages/QuizPage_L2.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    }, 300);
+  };
+
+  /**
+   * Sends the top three scores to the backend server.
+   * @param topScores The top three RIASEC scores to send.
+   */
+  const sendData = async (topScores: { type: RiasecType; score: number }[]) => {
+    try {
+      const res = await fetch("http://localhost:5001/api/quiz/filter", {
```
[the remark this answers is not in the document]
Comment (DamarisLi) [reply client/src/pages/QuizPage_L2.tsx]: Hier könnte man "http://localhost:5001" einfach weglassen, so wie in Zeile 126

**Comment by noadjamila** [comment]
Habe alles umgesetzt! @DamarisLi 
Schaust du noch einmal? 😅
